repos/system_upgrade/common/actors/yumconfigscanner/libraries/yumconfigscanner.py

```diff
@@ -0,0 +1,45 @@
+from leapp.libraries.common.config.version import get_source_major_version
+from leapp.libraries.stdlib import api, run
+from leapp.models import YumConfig
+
+
+def scan_enabled_yum_plugins():
+    """
+    Runs the `yum` command and parses its output for enabled/loaded plugins.
+
+    :return: A list of enabled plugins.
+    :rtype: List
+    """
+
+    # We rely on yum itself to report what plugins are used when it is invoked.
+    # An alternative approach would be to check /usr/lib/yum-plugins/ (install
+    # path for yum plugins) and parse corresponding configurations from
+    # /etc/yum/pluginconf.d/
+
+    if get_source_major_version() == '7':
+        yum_cmd = ['yum']
+    else:
+        yum_cmd = ['dnf', '-v']  # On RHEL8 we need to supply an extra switch
+
+    yum_output = run(yum_cmd, split=True, checked=False)  # The yum command will certainly fail (does not matter).
+
+    loaded_plugins = []
+    # Find the line that lists all the loaded plugins
+    for line in yum_output['stdout']:
+        if line.startswith('Loaded plugins:'):
```

Comment:
this could be broken as well. The  list could be split into the several lines so detection will not be possibly correct (see the [rhbz#1964311](https://bugzilla.redhat.com/show_bug.cgi?id=1964311) ticket where people reported this problem). I am not sure whether it happens only in the interactive console or always. But we should definitely find a time to check this behaviour to be sure. If this can happen during the upgrade, we need to find a more robust solution. E.g. check whether additional lines start with "\s+:\s" and in such a case process sequence of all these lines.
E.g.
```
# yum 2>&1|head -2
Loaded plugins: enabled_repos_upload, package_upload, product-id, search-
              : disabled-repos, subscription-manager
```
Pay the special attention to the split of the `search-disabled-repos` plugin.

Reply:
After a little bit of investigation about at what conditions precisely can the list of plugins be broken into multiple lines:
The `YUM` code responsible for handling this is [here](https://github.com/rpm-software-management/yum/blob/4ed25525ee4781907bd204018c27f44948ed83fe/yum/plugins.py#L222) (same code as present on RHEL7). To perform the formatting of the output they use the `textwrap.fill` function from **stdlib**. They do not modify the default values of the parameters responsible for configuring the text breaking, therefore:
- `break_long_words = True` implies that should some plugin have longer name than `termwidth` - some formatting prefix, the plugin name gets broken. 
- `break_on_hyphens = True` allows plugin names to be broken on hyphens (with hyphen being kept at the end of the line)
Furthermore we know, that the list of plugins is always separated by a `,`
Therefore, the parsing can be done as suggested above (checking for presence `\s+: `) to get the block of yum plugins, recreate the string containing the plugins (remove newlines and formatting prefixes, and concatenating the lines together) and breaking the result on `, `.

Reply:
This should be addressed in the latest version of this PR.